pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -235,19 +242,23 @@ func (plugin *cniNetworkPlugin) setDefaultNetwork(n *cniNetwork) {
 }
 
 func (plugin *cniNetworkPlugin) checkInitialized() error {
-	if plugin.getDefaultNetwork() == nil {
-		return errors.New("cni config uninitialized")
+	defaultCNINetwork := plugin.getDefaultNetwork()
+
+	if defaultCNINetwork == nil {
+		return fmt.Errorf("cni config uninitialized")
 	}
 
-	// If the CNI configuration has the ipRanges capability, we need a PodCIDR assigned
-	for _, p := range plugin.getDefaultNetwork().NetworkConfig.Plugins {
-		if p.Network.Capabilities["ipRanges"] {
-			if plugin.podCidr == "" {
-				return errors.New("no PodCIDR set")
-			}
-			break
-		}
+	netConf, cniNet := defaultCNINetwork.NetworkConfig, defaultCNINetwork.CNIConfig
+
+	caps, err := cniNet.ValidateNetworkList(context.TODO(), netConf)
```

Comment:
Yup, as @dcbw mentioned above, ValidateNetworkList should be called after we pick a CNI configuration, but before we mark ourselves as ready. That way, the user has the chance to fix any errors, but we don't flap and pick another config.

Reply:
What a coincidence! Could you help to answer the question above? @squeed